redash/handlers/dashboards.py

```diff
@@ -194,6 +213,11 @@ def get(self, dashboard_id=None):
 
         self.record_event({"action": "view", "object_id": dashboard.id, "object_type": "dashboard"})
 
+        # add allowed_widgets to the dashboard information to return in case it exists and it is not empty
+        allowed_widgets = get_allowed_widgets_info(dashboard_id)
+        if allowed_widgets:
+            response["allowed_widgets"] = allowed_widgets
```

Comment:
I'm not suggesting a change here, but I just wanna to talk more about strategies to have less intrusive changes on Redash. We can use the idea of merge conflicts as an heuristic of how "intrusive" our changes are. Here, for example, there are higher chances of merge conflicts since we're changing the view code itself. So, the question is: how could we avoid this?

I feel a good candidate here could be a Python decorator. By using it, we could keep the view code exactly the way it is and only change its signature to include a new decorator. This significantly reduces the change of, not only conflicts from happening, but also unexpected bugs. The decorator would decorate this view an update its response the same way it's doing here, but in a more isolated way;

For example, let's imagine that the Redash's main introduced changes to this view and, during the conflict resolution, for any reason we've done it wrongly and erased one of the new lines from Redash's main. This will create a bug and we'll be responsible to fix this.

But, again, no need to change anything. I'm only bringing up this conversation so we can consider such topics within newer PRs.

Reply:
thanks @berinhard I liked the decorator idea and tried its implementation [here](https://github.com/metr-systems/redash/pull/3/commits/02f9bc206383881d0c9b7a178e0612f25411fa0d)